src/raft.rs

```diff
@@ -709,9 +670,19 @@ impl<T: Storage> Raft<T> {
         let mut entry = Entry::new();
         entry.set_entry_type(EntryType::EntryConfChange);
         entry.set_data(data);
+
         // Index/Term set here.
-        self.append_entry(&mut [entry]);
+        if self.append_entry(&mut [entry]).is_err() {
+            panic!("{} append finalize entry shouldn't fail", self.tag);
+        }
+
         self.bcast_append();
+        if !self.prs().iter().any(|(id, _)| *id == self.id) {
+            self.become_follower(self.term, INVALID_ID);
+            debug!("{} append finalize entry and become follower", self.tag);
+            return;
+        }
+        debug!("{} append finalize entry", self.tag);
```

Comment:
Avoid the repeat log

```suggestion
        else { debug!("{} append finalize entry", self.tag); }
```